react/features/chat/components/native/ChatButton.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+
+import type { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+
+import {
+    AbstractButton,
+    type AbstractButtonProps
+} from '../../../base/toolbox';
+
+import { toggleChat } from '../../actions';
+import { getUnreadCount } from '../../functions';
+
+type Props = AbstractButtonProps & {
+
+    /**
+     * Display Chat.
+     *
+     * @protected
+     */
+    _onDisplayChat: Function,
+
+    /**
+     * The unread message count.
+     */
+    _unreadMessageCount: number
+};
+
+/**
+ * Implements an {@link AbstractButton} to open the chat screen on mobile.
+ */
+class ChatButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.chat';
+    iconName = 'chat';
+    label = '';
+    toggledIconName = 'chat1';
+
+    /**
+     * Handles clicking / pressing the button, and opens the appropriate dialog.
+     *
+     * @private
+     * @returns {void}
+     */
+    _handleClick() {
+        this.props._onDisplayChat();
+    }
+
+    /**
+     * Renders the button toggled when there are unread messages.
+     *
+     * @protected
+     * @returns {boolean}
+     */
+    _isToggled() {
+        return Boolean(this.props._unreadMessageCount);
```

Comment:
A part of me feels it's off to use a "toggled" state to show there are unread messages because I think of toggled as meaning off/on, but a majority of me finds it really clever and convenient.

Reply:
Yeah, I basically just reused an implemented way if state change of the ```AbstractButton``` (that is happened to be called "toggled" but coud be just "state_two" as well :) )